launch/launch/substitutions/string_join_substitution.py

```diff
@@ -76,6 +96,19 @@ def delimiter(self) -> List[Substitution]:
         """Getter for delimiter."""
         return self.__delimiter
 
+    @classmethod
+    def parse(cls, data: Sequence[SomeSubstitutionsType]):
+        """Parse `StringJoinSubstitution` substitution."""
+        if len(data) < 2:
+            raise TypeError(
+                'string-join substitution expects at least 2 arguments: '
+                '1 delimiter + at least 1 component'
+            )
+        kwargs: Dict[str, SomeSubstitutionsType | Iterable[SomeSubstitutionsType]] = {}
+        kwargs['delimiter'] = data[0]
+        kwargs['substitutions'] = data[1:]
+        return cls, kwargs
```

Comment:
After the change to the return type annotation, can you try to remove the type annotation for `kwargs` here and see if `mypy` is fine with it? Otherwise, you can try to just directly return a dict:

```suggestion
        return cls, {'delimiter': data[0], 'substitutions': data[1:]}
```

Reply:
Also included in 49dee17